### DISCUSSION TITLE
是否考虑采用go 1.18进行开发

### DISCUSSION THREAD

**Discussion by fanhousanbu** [go-atomci/atomci#105] [Ideas]
1.18正式版已出，目前使用的docker镜像是用了1.15，由于支持了`泛型`，是否可以考虑升级

弊端：这个泛型还是挺简陋的，对于接受器中使用泛型的支持也很糟糕

推荐原因：既然go不考虑方法重载，那么还有什么理由不选择泛型呢？

**Comment by sampsonye** [comment]
开源产品，在保证稳定的基础上，支持最新的技术实现想必是非常应该的。
其次，C#的泛型带来的生产力的提升是极为明显的（Java的伪泛型也是当下最频繁使用的技术之一）
最后，从官方Release和博客可以看到，有很多非常棒的提升：
https://golang.google.cn/doc/go1.18
https://golang.ir/blog/go1.18

性能、GC等等都有显著提升，因此我个人站队投**赞成票**

**Reply by colynn** [reply]
如果可以列举出哪一个实现的功能需求依赖此特性的，就支持升级， 

支持升级，但不建议直接使用最新版本，可以使用 1.16或是1.17

**Reply by fanhousanbu** [reply]
泛型非必须，但可以提高编程效率以及代码的精简，所以不存在“依赖”，但可以“更爽”；这也是之前并没有想讨论是否应该升级到1.16和1.17，但1.18却拿出来讨论的原因；
另外 也赞同 @sampsonye 的说法：开源产品，在可控的情况下，用最新技术是非常应该的
